state_transfer/endpoint/endpoint.go

```diff
@@ -4,23 +4,14 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const (
-	EndpointTypeRoutePassthrough  = "EndpointTypeRoutePassthrough"
-	EndpointTypeRouteInsecureEdge = "EndpointTypeRouteInsecureEdge"
-)
-
-type EndpointType string
-
 type Endpoint interface {
 	Create(client.Client) error
-	SetHostname(string)
 	Hostname() string
-	SetPort(int32)
 	Port() int32
 	Name() string
```

Comment:
Are these necessary for the user? Or could they be removed and only used by the implementation's that need them?

Reply:
@shawn-hurley I think those are needed for now, because of endpoint.Name() and endpoint.Namespace() is used a basis to derive all the object names later on as needed. Eventually, I think there will be a Metadata kind of struct that deals with name, namespace, labels, annotations, and owner references and is embedded in each object. This is one of the next things I will take on once this gets through. WDYT?

Reply:
If that is what it is used for, can we make this get NamesSpacedName() objects.NamespacedName or whatever? that way it is obvious they should be used together?

Reply:
updated.